solidity/test/e2e/Common.sol

```diff
@@ -3,14 +3,39 @@ pragma solidity =0.8.19;
 
 import {DSTestPlus} from '@defi-wonderland/solidity-utils/solidity/test/DSTestPlus.sol';
 import {IERC20} from 'isolmate/interfaces/tokens/IERC20.sol';
+import {SafeProxy} from 'safe-contracts/proxies/SafeProxy.sol';
+import {StorageMirror} from 'contracts/StorageMirror.sol';
+import {UpdateStorageMirrorGuard} from 'contracts/UpdateStorageMirrorGuard.sol';
+import {IGuardCallbackModule} from 'interfaces/IGuardCallbackModule.sol';
+import {IGnosisSafeProxyFactory} from 'test/e2e/IGnosisSafeProxyFactory.sol';
+import {TestConstants} from 'test/TestConstants.sol';
 
-contract CommonE2EBase is DSTestPlus {
+contract CommonE2EBase is DSTestPlus, TestConstants {
   uint256 internal constant _FORK_BLOCK = 15_452_788;
 
-  address internal _user = makeAddr('user');
-  address internal _owner = makeAddr('owner');
+  address public deployer = makeAddr('deployer');
+  address public safeOwner = makeAddr('safeOwner');
+  address public proposer = makeAddr('proposer');
 
-  function setUp() public {
+  StorageMirror public storageMirror;
+  UpdateStorageMirrorGuard public updateStorageMirrorGuard;
+  SafeProxy public safe;
```

Comment:
I would save the safe as an `ISafe` and cast the proxy into it when you deploy it, as the proxy doesnt have any functions and only uses fallbacks